monix-tail/shared/src/main/scala/monix/tail/Iterant.scala

```diff
@@ -1874,6 +1878,12 @@ sealed abstract class Iterant[F[_], A] extends Product with Serializable {
   *         the stream when evaluated
   */
 object Iterant extends IterantInstances {
+  // @TODO move somewhere else?
+  sealed trait BracketResult
+  case object Completed extends BracketResult
+  case object EarlyStop extends BracketResult
+  case class Error(e: Throwable) extends BracketResult
+
```

Comment:
I was going to do something similar with typelevel/cats-effect#113. Tho here for `BracketResult`:
* The error type parameter is not needed, since `Iterant` is bound to `Throwable`
* Signaling the result in `Completed` probably does not make sense:
  * It's not possible to always signal a `B` from `use` because it might return an empty Iterant
  * It's possible to signal a `Seq[B]` (which would potentially require unlimited buffering, I'd rather not have an operator do that implicitly) or an `Option[B]` with the first element (for which I could not find any compelling use case)

Reply:
Indeed, we don't have a result for `Completed`. This type should be top-level in `monix.tail`.

But make `BracketResult` a `sealed abstract class` and the `case` items members of `object BracketResult`. Oh and that `case class` needs to be `final`. We think of them as `final`, but the Scala compiler doesn't do it for legacy reasons.